### ISSUE TITLE
Message bubbles: make replies clearer by putting the quoted message above the reply

### ISSUE THREAD

**Issue by frandavid100** [vector-im/element-web#18171]
I think that replies would look clearer in the bubbled messages interface if the quoted message didn't appear inside the reply, and instead appeared above it. Here's a mockup showing how they look now vs how they could look:

![image](https://user-images.githubusercontent.com/694794/126603293-a1b11006-2f59-4686-9382-0ec89804719c.png)

In my mockup, an extract of Germain's original message appears above NightShade's message, which is a reply. Germain's message takes as little space as possible by only showing the avatar and the first line of the message. It also has a different background color, and there's a vertical line threading it to the reply. When clicked it takes you to Germain's actual message, which is a little further up.

**Comment by ShadowJonathan** [comment]
When doing this, I'd suggest following somewhat discord's footsteps, here's an example;

![image](https://user-images.githubusercontent.com/22740616/126604202-b04b6ca5-0fc4-4688-a0b6-72c20ce6e15e.png)

Most notably, discord insets the reply and makes the replied-to's avatar smaller.

**Comment by SimonBrandner** [comment]
Related: https://github.com/vector-im/element-web/issues/18145

**Comment by frandavid100** [comment]
> When doing this, I'd suggest following somewhat discord's footsteps, here's an example;
> 
> ![image](https://user-images.githubusercontent.com/22740616/126604202-b04b6ca5-0fc4-4688-a0b6-72c20ce6e15e.png)
> 
> Most notably, discord insets the reply and makes the replied-to's avatar smaller.
> 
> In general though, i'm favour of this approach, it feels better-virtualized in a bubble-message environment.

Hi, Jonathan. I'm glad you like this approach.

I'd rather not indent the replied-to message. If we don't indent it, we can do this when it's right above the reply:

![image](https://user-images.githubusercontent.com/694794/126605125-f27077e8-2d60-41df-9aca-9627594fc531.png)

In this mockup, there is no message between Germain's message and NightShade's reply. Therefore, their relation can be represented simply by threading them directly with a vertical line. This saves some space and, in my opinion, makes the timeline simpler.

**Comment by ShadowJonathan** [comment]
That'd introduce a bit of complexity that would not be often shown (as i dont think many people will reply to a message that was sent just now, they'd rather "cross the bridge" by replying to something sent a few messages up), the problem with that is that *if* it shows up, a user might be confused and wouldn't "click" with the new design, or know where their expected reply indication went.

I'd argue to be consistent, even when the message is directly above, as that de-complicates code and makes the reply feature uniform and "reliable" (in the user's eyes).

**Comment by kojid0** [comment]
What about this design?

![github](https://user-images.githubusercontent.com/24477557/126606669-14732564-484e-4a49-9435-0f76016ecdbd.png)

**Comment by SimonBrandner** [comment]
> What about this design?
> 
> ![github](https://user-images.githubusercontent.com/24477557/126606669-14732564-484e-4a49-9435-0f76016ecdbd.png)

Yeah, that's very similar to what we have on develop for the default layout atm

**Comment by frandavid100** [comment]
> I dont think many people will reply to a message that was sent just now

Yup, that's probably the situation now. But when Element starts using threading, I think that many people will reply to even very recent messages, in order to keep conversations grouped. If that turns out to be the case, we might want the interface to be as comfortable to the user as possible even with a big amount of replies.

For example, if Element used threading, Simon might have wanted to reply to Nightshade to keep all inside the same thread. In that case, the whole thread could look like this:

![image](https://user-images.githubusercontent.com/694794/126607185-92cf5c55-63c7-4b71-a993-769a4a78e505.png)

I'm not saying we _need_ to introduce this whole threading interface now. I'm just saying that, if we end up needing it when Element starts to do threads, it will look more familiar if we don't indent the replied-to message now.

**Comment by SimonBrandner** [comment]
> For example, if Element used threading, Simon might have wanted to reply to Nightshade to keep all inside the same thread. In that case, the whole thread could look like this:

That wouldn't be the way to do threading, you'd probably have a separate view for the thread, this would turn very chaotic very quickly

**Comment by frandavid100** [comment]
> That wouldn't be the way to do threading, you'd probably have a separate view for the thread

Thanks, is there somewhere I can find out more about plans for threading in Element? Right now, I assume:

1. That replying to a message starts a thread, and puts that reply in that thread.
2. That messages in a thread appear on a separate view, but can also appear in the main timeline.

Is that not the case?

**Comment by ShadowJonathan** [comment]
Replying to a message doesn't start a thread, threads are dedicated and seperate from replying (though i don't know if Product wants to coerce both into one).

**Comment by SimonBrandner** [comment]
> is there somewhere I can find out more about plans for threading in Element?

I don't know of anything, sorry

**Comment by ShadowJonathan** [comment]
Threading is still in spec land, while some designs have flown around on how it'll look, nothing's definitive. Here's the spec link though; https://github.com/matrix-org/matrix-doc/pull/2836

**Comment by frandavid100** [comment]
> Replying to a message doesn't start a thread, threads are dedicated and seperate from replying (though i don't know if Product wants to coerce both into one).

Kind of off-topic, but is there a reason why you wouldn't want replies to start a thread? From my point of view, "reply to this message" and "start a thread from this message" serve basically the same purpose. Why would they be two separate things?

Again, since this is off-topic here. Is there a better place to have this conversation?

**Comment by SimonBrandner** [comment]
> Kind of off-topic, but is there a reason why you wouldn't want replies to start a thread? From my point of view, "reply to this message" and "start a thread from this message" serve basically the same purpose. Why would they be two separate things?

Personal preference - I prefer not to thread my DMs as I feel like threading is an overkill for my casual DMs

> Again, since this is off-topic here. Is the spec a better place to have this conversation?

Spec isn't a good place for this as this is a UI/UX issue

**Comment by ShadowJonathan** [comment]
> Kind of off-topic, but is there a reason why you wouldn't want replies to start a thread

Then you get slack's approach, where everything is a thread. I don't like that approach, as that banishes replies to the sidebar (unless you "stick a foot in the door" by also having the message appear in the timeline, explicitly).

**Comment by frandavid100** [comment]
> Personal preference - I prefer not to thread my DMs as I feel like threading is an overkill for my casual DMs

Even if threads are created simply by replying, and messages in threads can appear on the timeline, and they look exactly like regular replies?

> Spec isn't a good place for this as this is a UI/UX issue

What would be a better place, then?

**Comment by ShadowJonathan** [comment]
> What would be a better place, then?

I'd recommend `#element-design:matrix.org`, though that one is almost underused.

**Comment by frandavid100** [comment]
> I'd recommend `#element-design:matrix.org`, though that one is almost underused.

Thanks!

> Then you get slack's approach, where everything is a thread. I don't like that approach, as that banishes replies to the sidebar (unless you "stick a foot in the door" by also having the message appear in the timeline, explicitly).

That is exactly what I'd want, except:

1. Users would be able to silence threads in order to hide their messages from their timeline, or unsilence them in order to show them, or even subscribe to them in order to get notifications. That would give them a great amount of control about what kind of content they see, even in big rooms.
2. Room administrators could set a default, e.g. silence threads by default if they want a slack-like experience.
3. Clients could also set defaults based on several factors, e.g. subscribe by default in DMs.

Do you think that would be OK?

**Comment by ShadowJonathan** [comment]
> Do you think that would be OK?

I don't, for various UX reasons, but that's a difference of opinion, and I'll not be saying which approach is right. In the end, I'll leave Design and Product to decide that, I think they have enough in this thread to work from and decide on internally.

**Comment by frandavid100** [comment]
Thanks for the reply anyway 🙂

**Comment by vertigo220** [comment]
I like the proposed change, which I think makes things a bit more readable, but I have a couple thoughts:

- I like making the quoted text lighter so it's obvious it's not a "normal" part of the conversation, but I'm not sure about making it smaller. This is partly because every time you'd want to read the full message (probably not often, but could be), that's an extra step, but mostly because I'd want clicking on it to expand it, since I wouldn't want to be thrown up in the conversation then have to find my way back. Though I do like being able to do so, but that should probably be via a separate button/icon.
- As for threading, I think a possible good way to do this would be to simply add an option to messages to view their thread, at which point it could just display all the messages tied to that, which it could determine by looking at all the quotes/replies. For example, message 1 is sent, then message 2, then message 3 is a reply to message 1, then message 4 is sent, then message 5, then message 6 is a reply to message 3, and so on. If someone were to click on any of messages 1, 3, or 6 and select "View thread," it would show all of them because they're linked through replies.
